From 4674817cff47369fbfccc6211f6731a50fbc11dc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 06 May 2019 19:00:06 +0300
Subject: [PATCH] [SW-0] pll: fix long long goya boot

Commit:
3a9af9c ([SW-0] pll: refine the init sequence)
while did the refinement of the PLL init sequence, moved the
"wait for lock" code to run after the divisors init code, but forgot
to move the hl_pll_clear_lock_status() function which is essential
for Goya to be run after the the lock status is signaled.

This resulted in the relevant lock bit got cleared before the
"wait for lock" has been performed and thus the new
hl_pll_wait_for_lock() function has exhausted the timeout for each PLL.

Move the hl_pll_clear_lock_status() function invocation to after the
hl_pll_wait_for_lock() function call.

Fixes: 3a9af9c76d82369a4fb4dd9ea250456859b65f5f

Change-Id: I7f2c229861f02ad9e7257a57b40af446dcef9219
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 86c05c7..d4cb726 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -68,9 +68,6 @@
 	/* Wait >500 (10us) reference cycles until PLL re-locks. */
 	hl_pll_udelay(HL_PLL_MAX_WAIT_TIME);
 
-	/* w/a the h/w erratum: lock status bit isn't getting cleared */
-	hl_pll_clear_lock_status(cfg);
-
 	return 0;
 }
 
@@ -153,6 +150,9 @@
 	/* to be on a safe side, poll on the PLL lock status bit */
 	hl_pll_wait_for_lock(cfg);
 
+	/* w/a the h/w erratum: lock status bit isn't getting cleared */
+	hl_pll_clear_lock_status(cfg);
+
 	hl_pll_ack_int(cfg);
 
 	hl_pll_bypass(cfg, false);
